packages/silverengine/core/src/Engine/CLI.php

```
@@ -50,12 +50,17 @@ private function optimize(): void
         $cacheDir = ROOT . 'storage/cache/';
         @mkdir($cacheDir, 0775, true);
 
-        // Route cache: include the route files to populate Route, then
-        // dump flat definitions — unless a Closure route forbids it.
-        foreach (Env::get('routes', []) as $route) {
-            include_once ROOT . $route . '.php';
+        // Route cache: include the route files to populate the router,
+        // then dump flat definitions — unless a Closure route forbids it.
+        $router = app(Route::class);
+        $router->reset();
+        $loadRouteFile = static function (string $path, Route $route): void {
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$route'. (undefined)

(UnusedFormalParameter)